frontend/appflowy_flutter/lib/startup/tasks/generate_router.dart

```diff
@@ -0,0 +1,270 @@
+import 'package:appflowy/mobile/presentation/presentation.dart';
+import 'package:appflowy/startup/startup.dart';
+import 'package:appflowy/startup/tasks/app_widget.dart';
+import 'package:appflowy/user/application/auth/auth_service.dart';
+import 'package:appflowy/user/presentation/presentation.dart';
+import 'package:appflowy/workspace/presentation/home/desktop_home_screen.dart';
+import 'package:flowy_infra/time/duration.dart';
+import 'package:flutter/material.dart';
+import 'package:go_router/go_router.dart';
+
+GoRouter generateRouter(Widget child) {
+  return GoRouter(
+    navigatorKey: AppGlobals.rootNavKey,
+    initialLocation: '/',
+    routes: [
+      GoRoute(
+        path: '/',
+        // Root route is SplashScreen.
+        // It needs LaunchConfiguration as a parameter, so we get it from ApplicationWidget's child.
+        pageBuilder: (context, state) => MaterialPage(
+          child: child,
+        ),
+      ),
+      // Routes in both desktop and mobile
+      GoRoute(
+        path: SignInScreen.routeName,
+        pageBuilder: (context, state) {
+          return CustomTransitionPage(
+            child: SignInScreen(router: getIt<AuthRouter>()),
+            transitionsBuilder: _buildFadeTransition,
+            transitionDuration: Duration(
+              milliseconds: (RouteDurations.slow.inMilliseconds).round(),
+            ),
+          );
+        },
+      ),
+      GoRoute(
+        path: SkipLogInScreen.routeName,
+        pageBuilder: (context, state) {
+          return CustomTransitionPage(
+            child: SkipLogInScreen(
+              authRouter: getIt<AuthRouter>(),
+              authService: getIt<AuthService>(),
+            ),
+            transitionsBuilder: _buildFadeTransition,
+            transitionDuration: Duration(
+              milliseconds: (RouteDurations.slow.inMilliseconds).round(),
+            ),
+          );
+        },
+      ),
+      GoRoute(
+        path: EncryptSecretScreen.routeName,
+        pageBuilder: (context, state) {
+          final args = state.extra as Map<String, dynamic>;
+          return CustomTransitionPage(
+            child: EncryptSecretScreen(
+              user: args['user'],
+              key: args['key'],
+            ),
```

Comment:
Please don't hardcode the keys. It will be better to use class or const string instead.

Reply:
I changed it to const since all we need is to avoid typos in this case.